### PR TITLE
add some tests in fs test

### PR DESCRIPTION
**Pull request by BolaNasr**

### CODE REVIEW

File: tests/sals/fs/test_fs.py [hunk 1]
```diff
@@ -9,6 +13,8 @@ class TestFS(BaseTests):
     def setUpClass(cls):
         cls.temp_path = j.sals.fs.join_paths("/tmp", cls.generate_random_text())
 
+    def random_string(self):
+        return str(uuid4())[:10]
```
Comment (abom) [inline tests/sals/fs/test_fs.py:17]: You can use `j.data.random_names` or `j.data.idgenerator` directly?

File: tests/sals/fs/test_fs.py [hunk 2]
```diff
@@ -157,3 +163,178 @@ def test012_walk_dirs(self, recursive):
...
+        #. Write binary file.
+        #. Check that file has been created and it is a binary file.
+        #. Read this file and check that its content.
+        """
+        self.info("Write binary file.")
+        content = self.random_string()
+        random_dir_dest, random_dir_dest_2, _, _ = self.create_tree()
+        random_dir_dest_3 = j.sals.fs.join_paths(random_dir_dest_2, self.generate_random_text())
+        with open(random_dir_dest_3, "wb") as f:
+            pickle.dump(content, f)
```
Comment (abom) [inline tests/sals/fs/test_fs.py:199]: Why do we use pickle here? we can write any binary content? and I think the implementation of binary file in this sal just means non-ascii.

File: tests/sals/fs/test_fs.py [hunk 3]
```diff
@@ -157,3 +163,178 @@ def test012_walk_dirs(self, recursive):
...
+
+
+    def test_0018_change_file_names(self):
+        """
+        Test case for changing files names using j.sal.fs.change_filenames.
+        """
+        self.info("Create a tree with many sub directories and files with a common word (W1) in their names.")
+        base_dir,_,_,_ = self.create_tree()
+        
+        self.info("Change these files names by replacing (W1) with another word (W2) with recursive=False.")
```
Comment (AhmedHanafy725) [inline tests/sals/fs/test_fs.py:265]: `change_names` doesn't have recursive argument.

File: tests/sals/fs/test_fs.py [hunk 4]
```diff
@@ -157,3 +163,178 @@ def test012_walk_dirs(self, recursive):
...
+        log_files = [
+            os.path.splitext(x)[0]
+            for x in  j.sals.fs.walk_files("/tmp")
```
Comment (AhmedHanafy725) [inline tests/sals/fs/test_fs.py:268]: Please use `self.temp_path` instead of `"/tmp"`

Comment (AhmedHanafy725) [reply tests/sals/fs/test_fs.py:268]: Does the tree has been created has `.log` files

File: tests/sals/fs/test_fs.py [hunk 5]
```diff
@@ -157,3 +163,178 @@ def test012_walk_dirs(self, recursive):
...
+        
+        self.info("Check that children files are only changed.")
+        changed_files = [
+            os.path.splitext(x)[0]
+            for x in j.sals.fs.walk_files("/tmp")
+            if ".java" in x
+        ]
+        assert changed_files == child_log
+        
+        self.info("Change these files names again by replacing (W1) with another word (W2) with recursive=True.")
```
Comment (AhmedHanafy725) [inline tests/sals/fs/test_fs.py:282]: The following scenario is repeated.

File: tests/sals/fs/test_fs.py [hunk 6]
```diff
@@ -157,3 +163,178 @@ def test012_walk_dirs(self, recursive):
...
+        ]
+        assert sorted(log_files) == sorted(java_files)
+
+        self.info("Create a new .py file")
+
+        file_name = f"{self.random_string()}.py"
+        file_path = os.path.join(base_dir, file_name)
+        j.sals.fs.touch(file_path)
+        
+        self.info("Change the files names with modification time not more than 2 seconds ago.")
```
Comment (AhmedHanafy725) [inline tests/sals/fs/test_fs.py:299]: Changing files names with modification time is not supported here, no?

File: tests/sals/fs/test_fs.py [hunk 7]
```diff
@@ -157,3 +163,178 @@ def test012_walk_dirs(self, recursive):
...
+        # j.sals.fs.change_filenames(".py", ".html", base_dir) need to implement it .
+
+        files_python = [
+            x
+            for x in j.sals.fs.walk_files(base_dir)
+            if ".py" in x
+        ]
+        assert len(files_python) == 1
+
+    def test_019_change_name_files(self):
```
Comment (AhmedHanafy725) [inline tests/sals/fs/test_fs.py:310]: I didn't get the point of the test

**Review by AhmedHanafy725 — COMMENTED** [review]
docstring are missing for some tests.

**Review by AhmedHanafy725 — COMMENTED** [review]
Black formatting is failing, can you check it?

File: tests/sals/fs/test_fs.py [hunk 8]
```diff
@@ -157,3 +161,157 @@ def test012_walk_dirs(self, recursive):
         else:
             self.assertEqual(1, len(dirs))
             self.assertNotIn(random_dir_dest_3, dirs)
+
+    def test013_path_parts(self):
```
Comment (waleedhammam) [inline tests/sals/fs/test_fs.py:165]: should we add docstring here ?

File: tests/sals/fs/test_fs.py [hunk 9]
```diff
@@ -157,3 +161,157 @@ def test012_walk_dirs(self, recursive):
...
+        random_dir_dest_3 = j.sals.fs.join_paths(self.temp_path, self.generate_random_text())
+        self.info("Create dir {}".format(random_dir_dest_3))
+        parts = list(j.sals.fs.path_parts(random_dir_dest_3))
+        split_path = random_dir_dest_3.split("/")
+        split_path[0] = "/"
+        self.assertEqual(parts, split_path)
+
+    def test014_basename(self):
```
Comment (waleedhammam) [inline tests/sals/fs/test_fs.py:173]: also in here

File: tests/sals/fs/test_fs.py [hunk 10]
```diff
@@ -157,3 +161,157 @@ def test012_walk_dirs(self, recursive):
         else:
             self.assertEqual(1, len(dirs))
             self.assertNotIn(random_dir_dest_3, dirs)
+
+    def test013_path_parts(self):
+        random_dir_dest_3 = j.sals.fs.join_paths(self.temp_path, self.generate_random_text())
+        self.info("Create dir {}".format(random_dir_dest_3))
```
Comment (waleedhammam) [inline tests/sals/fs/test_fs.py:167]: does it actually creates the dir or just the path string ?

**Pull request closed without merging**